modules/rte_mis_allocation/src/Access/AllocationDetailEditAccessCheck.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\rte_mis_allocation\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\eck\EckEntityInterface;
+use Drupal\user\UserInterface;
+
+/**
+ * Determines edit access to the school details mini node.
+ */
+class AllocationDetailEditAccessCheck implements AccessInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The config factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  public $configFactory;
+
+  /**
+   * Constructs an UserRegisterAccessCheck object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Config factory.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, ConfigFactoryInterface $config_factory) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->configFactory = $config_factory;
```

Comment:
if we are not using `$this->configFactory`, then please remove this.

Reply:
ya sure we can remove this.